### ISSUE TITLE
Yetus should be accessible outside pull requests

### ISSUE THREAD

**Issue by cshari-zededa** [lf-edge/eve#183]
Please implement a way to run Yetus checks on our code changes before we create a pull request. Right now, we submit a PR, and then come to know about Yetus errors as part of the checks.  We end up wasting circleCI bandwidth in iterating over Yetus failures, and becomes a time hurdle in code reviews.

What would be good is to have a way to run Yetus on our local EVE repository, so that we can validate Yetus in our laptop before we even create a PR.

**Comment by ckzed** [comment]
One easy workaround is to install the tools that yetus uses internally (revive, shellcheck, hadolint, pylint) in your development environment. There are editor plugins, modules, packages which can be added which show one the errors/warnings as you type.

Regards,
Chirag Kantharia.

**Comment by cshari-zededa** [comment]
This approach will not work. How do we make sure everybody runs the same set of checks and they are in sync with the Yetus configuration?

Thanks
Hari

**Comment by rvs** [comment]
@cshari-zededa two points, first of all, you can run yetus easily on your own. Just follow the same steps outlined in our .circleci config yaml https://github.com/lf-edge/eve/blob/master/.circleci/config.yml#L46

Also, personally I have no problems with using PR mechanism to run yetus -- sure your first PR is likely to fail, but that's not that big of a deal and you can always follow up with a change that takes care of yetus feedback. Personally -- even though I can run yetus locally -- I'm tool lazy to do that and just use PR mechnism.

**Comment by deitch** [comment]
I can sympathize here. If I have to wait for a PR test run to report success or failure, it is going to slow down dev cycles. That has led to my personal preference (not always realistic, but preference nonetheless) for my CI config files to always do `make <something>`, so that I can run the same thing locally just as easily.

Can we perhaps have a docker image built that contains everything yetus needs (auto-correct keeps changing that to "fetus", which is odd - "contains everything fetus needs"?), so that both circleci and a normal user can just run `make ci` or even `make yetus` and get the exact same output quickly?

**Comment by rvs** [comment]
> I can sympathize here. If I have to wait for a PR test run to report success or failure, it is going to slow down dev cycles. That has led to my personal preference (not always realistic, but preference nonetheless) for my CI config files to always do make <something>, so that I can run the same thing locally just as easily.

I actually don't get it: for me this is the other way around -- PRs are cheap and fire-and-forget. I typically work on a few issues in parallel anyway so while one PR is testing I can spend time doing something else. In fact, this is much easier than to try and run things locally. Once yetus posts its feedback on a PR -- fixing it is typically super straightforward as well.

> Can we perhaps have a docker image built that contains everything yetus needs (auto-correct keeps changing that to "fetus", which is odd - "contains everything fetus needs"?), so that both circleci and a normal user can just run make ci or even make yetus and get the exact same output quickly?

We already have that container and how to run it specified in a circleci config. If you want to hook it up to Makefile -- sure that'd be fine, but personally I have no need for it so it'll have to be one of you guys actually doing the work ;-)

**Comment by deitch** [comment]
> I actually don't get it: for me this is the other way around -- PRs are cheap and fire-and-forget. I typically work on a few issues in parallel anyway so while one PR is testing I can spend time doing something else

Different strokes for different folks and all that. :-)

**Comment by deitch** [comment]
> We already have that container and how to run it specified in a circleci config. If you want to hook it up to Makefile -- sure that'd be fine, but personally I have no need for it so it'll have to be one of you guys actually doing the work ;-)

Consider it done. If it makes life easier for some people, happy to see it done. See #185

**Comment by zach-welch-timesys** [comment]
FWIW, a `make yetus` rule got added when #195 was merged, so this issue may deserve to be closed. There may be some bugs with the implementation, but those probably should be filed as new issues.

**Comment by deitch** [comment]
Agreed. If there are bugs, they should be under new issues. If the way to run it isn’t sufficient, then we need a new issue describing such improvement. 

Closing.
